src/backends/r1cs/mod.rs

```diff
@@ -478,15 +479,80 @@ where
             let (line, _, line_str) = crate::utils::find_exact_line(source, *span);
             let line_str = line_str.trim_start();
             let dbg_msg = format!("[{filename}:{line}] `{line_str}` -> ");
-            for cvar in var_info.var.iter() {
-                match cvar {
-                    ConstOrCell::Const(cst) => {
-                        println!("{dbg_msg}{}", cst.pretty());
+            
+            match &var_info.typ {
+                // Field
+                Some(TyKind::Field { .. }) => {
+                    match &var_info.var[0] {
+                        ConstOrCell::Const(cst) => {
+                            println!("{dbg_msg}{}", cst.pretty());
+                        }
+                        ConstOrCell::Cell(cell) => {
+                            let val = cell.evaluate(&witness);
+                            println!("{dbg_msg}{}", val.pretty());
+                        }
                     }
-                    ConstOrCell::Cell(cell) => {
-                        let val = cell.evaluate(&witness);
-                        println!("{dbg_msg}{}", val.pretty());
+                }
+
+                // Bool
+                Some(TyKind::Bool) => {
+                    match &var_info.var[0] {
+                        ConstOrCell::Const(cst) => {
+                            let val = *cst == F::one();
+                            println!("{dbg_msg}{}", val);
+                        }
+                        ConstOrCell::Cell(cell) => {
+                            let val = cell.evaluate(&witness) == F::one();
+                            println!("{dbg_msg}{}", val);
+                        }
+                    }
+                }
+
+                // Array
+                Some(TyKind::Array(_, _)) => {
```

Comment:
Also it could be recursive array.

For example 
1. [[1, 2],[3, 4]]
2. [thing1, thing2] // thing* are structs

So maybe it would be better to recursively print out the elements based on the element's type.

Reply:
Log Array Array Field.
![array_array_field](https://github.com/user-attachments/assets/e8408bc0-f521-41f9-971a-d1900deb16d3)


Log Array Array Bool.
![array_array_bool](https://github.com/user-attachments/assets/7b2b46ad-cdb1-4920-b08c-a79af4f4ac03)


Log Array Custom.
![array_custom](https://github.com/user-attachments/assets/d7a4b2d5-e080-4bba-bdb5-9ca3911e6fc2)